### ISSUE TITLE
Remove opencv dependency from tests

### ISSUE THREAD

**Issue by guarin** [lightly-ai/lightly#1477]
We still have an old opencv dependency in the tests, see here: https://github.com/lightly-ai/lightly/blob/2b215aa5d19b50515f280c1b0181f2f135cbc93f/tests/data/test_VideoDataset.py#L11

We should refactor the test(s) to not rely on opencv and remove opencv from all requirements files.

**Comment by ishaanagw** [comment]
Hey @guarin, Can I work on this ?

**Comment by SauravMaheshkar** [comment]
Sure @ishaanagw give it a go 😃

**Comment by ishaanagw** [comment]
Can you please explain the task a little more please ? Do I have to replace the videoWriters from OpenCV to some other library ?

**Comment by guarin** [comment]
> Can you please explain the task a little more please ? Do I have to replace the videoWriters from OpenCV to some other library ?

Yes the idea would be to replace `cv2.VideoWriter` with [`write_video`](https://pytorch.org/vision/main/generated/torchvision.io.write_video.html) from torchvision. Looking at it this could actually be a more tricky issue than anticipated. If you would like to work on it, could you try the following:
* Replace all `cv2.VideoWriter` with torchvision `write_video` calls instead
* Replace the numpy arrays containing the frames (see [here](https://github.com/lightly-ai/lightly/blob/2b215aa5d19b50515f280c1b0181f2f135cbc93f/tests/data/test_VideoDataset.py#L53-L56)) with tensors because `write_video` expects tensors
* Try to run the tests with `pytest tests/data/test_VideoDataset.py`

If you run into errors regarding video timestamps we most likely can't remove cv2.

**Comment by ishaanagw** [comment]
Getting this error -  ValueError: Expected numpy array with ndim `3` but got `2`
in self._test_video_dataset_non_increasing_timestamps()

Is this what you were talking about ?

**Comment by guarin** [comment]
Interesting, could you post the full stack trace? 

> Is this what you were talking about ?

No I believe the ValueError already happens before. With the timestamps I meant that depending on which library is used when creating a new video file, the timestamps stored in the file can be different. And I am not sure if `write_video` creates the exact same timestamps as `cv2.VideoWriter`.

**Comment by ishaanagw** [comment]
tests/data/test_VideoDataset.py:238: in _test_video_dataset_non_increasing_timestamps
    self.create_dataset(n_videos=2, n_frames_per_video=5)
tests/data/test_VideoDataset.py:83: in create_dataset
    out = torchvision.io.write_video(path, frame, n_frames_per_video)
../miniconda3/envs/lightly/lib/python3.10/site-packages/torchvision/io/video.py:131: in write_video
    frame = av.VideoFrame.from_ndarray(img, format="rgb24")
av/video/frame.pyx:551: in av.video.frame.VideoFrame.from_ndarray

**Comment by guarin** [comment]
This looks like something is wrong with the dimensions of the numpy arrays. Each frame should have dimension `(W, H, C)` with `C = 3`. But somehow the array it got has only 2 dimensions instead of 3.

**Comment by SGCODEX** [comment]
Can I work on this issue as well @guarin

**Comment by ishaanagw** [comment]
I think we can remove it, I was able to remove cv2 dependencies but facing some test failure, these test fail with no changes as well. 

Failed Test Cases - 
FAILED tests/data/test_VideoDataset.py::TestVideoDataset::test_video_dataset_dataloader__pyav - TypeError: cannot pickle '_thread.lock' object
FAILED tests/data/test_VideoDataset.py::TestVideoDataset::test_video_dataset_non_increasing_timestamps__pyav - TypeError: cannot pickle 'VideoLoader' object

Do you think this is due to some local setup ?

**Comment by moli-debugger** [comment]
I would like to pick this up as well @guarin

**Comment by moli-debugger** [comment]
@SGCODEX , please pick up this task . I will continue work on the other issues.

CC: @guarin

**Comment by ishaanagw** [comment]
> > Do you think this is due to some local setup ?
> 
> Could be, which operating system are you using? Mac OS
 You can also already create a PR and push it, then we can check if it is due to the local setup or not.
> Create PR for this - https://github.com/lightly-ai/lightly/pull/1678
